### PR TITLE
WT-6224 Adding globally visible tombstone when on disk prepare is rol…

### PR DESCRIPTION
**Pull request by luke-pearson**
…led back and no associated history store entry exists

### CODE REVIEW

File: src/txn/txn.c [hunk 1]
```diff
@@ -863,7 +861,7 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
         /*
          * Newer updates are inserted at head of update chain, and transaction operations are added
          * at the tail of the transaction modify chain.
-         *
+         *-
```
Comment (quchenhao) [inline src/txn/txn.c:864]: Remove -

**Comment by luke-pearson** [comment]
Accidentally added some extra stuff to my PR, don't review yet please.

File: src/txn/txn.c [hunk 2]
```diff
@@ -895,8 +893,33 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
...
+                cbt = (WT_CURSOR_BTREE *)*cursorp;
+
+                /*
+                 * Allocate a tombstone so that when we reconcile the update chain we don't copy the
+                 * prepared cell, which is now associated with a rolled back prepare, and instead
+                 * write nothing.
+                 */
+                WT_RET(__wt_upd_alloc_tombstone(session, &tombstone, &size));
+                /* Apply the tombstone to the row. */
+                WT_WITH_BTREE(session, op->btree, ret = __wt_row_modify(cbt, &cbt->iface.key, NULL,
```
Comment (quchenhao) [inline src/txn/txn.c:914]: This is wrong, I think. We can only append at the end of the update chain not at head. See WT-6228

Comment (luke-pearson) [reply src/txn/txn.c:920]: I think once WT-6235 goes in I'll move it to be ahead of the point at which we abort the update, then I don't think we can race. If you think we still could, please explain how.

Comment (quchenhao) [reply src/txn/txn.c:920]: May be that's safe, but I still want to append at the end. It feels natural to me as the updates do occur before the prepared updates and it is potentially safer. Imagine there is already a reader on the update chain, if we append at the end, we can reduce its chance to restart the read again.

Comment (luke-pearson) [reply src/txn/txn.c:920]: Conceptually it makes more sense to me that we put it ahead of the prepared update, in essence deleting everything that comes after it. I think the situation being discussed isn't likely to occur enough to justify appending to the end. In regards to updates occurring prior to the prepared update, we know that in this scenario that isn't possible.

Comment (quchenhao) [reply src/txn/txn.c:920]: I think we can't agree on this. @bvpvamsikrishna @hbokhari What do you guys think?

Comment (bvpvamsikrishna) [reply src/txn/txn.c:920]: terms of before and after are confusing for me. We need to add tombstone at the head of the update chain, considering the newest will be at the head. This is the natural order.

**Comment by luke-pearson** [comment]
Closing in favour of: #5685

**Pull request closed without merging**